src/main/java/players/mcts/SingleTreeNode.java

```diff
@@ -557,13 +560,14 @@ protected void advance(AbstractGameState gs, AbstractAction act, boolean inRollo
      */
     protected void advanceToTurnOfPlayer(AbstractGameState gs, int id, boolean inRollout) {
         // For the moment we only have one opponent model - that of a random player
+        AbstractAction action = null;
         while (gs.getCurrentPlayer() != id && gs.isNotTerminalForPlayer(id) && !(inRollout && finishRollout(gs))) {
             //       AbstractGameState preGS = gs.copy();
             AbstractPlayer oppModel = opponentModels[gs.getCurrentPlayer()];
-            List<AbstractAction> availableActions = forwardModel.computeAvailableActions(gs);
+            List<AbstractAction> availableActions = forwardModel.computeAvailableActions(gs, params.actionSpace);
             if (availableActions.isEmpty())
-                throw new AssertionError("Should always have at least one action possible...");
-            AbstractAction action = oppModel.getAction(gs, availableActions);
+                throw new AssertionError("Should always have at least one action possible..." + (action != null? " Last action: " + action : ""));
+            action = oppModel.getAction(gs);
```

Comment:
I'm slightly uncomfortable with this. 
First we calculate all the actions...and then ignore them when we get the actual action from the player (or oppModel). This is very inefficient.

We should just generate the agents once. In line with the proposals in the Game class; either we never generate the actions here, and rely on the Player doing this correctly; OR we get the ActionSpace/FM the player is using and generate the cations here (for error checking), before passing the list to the agent to make a decision, without recalculating everything.